wurst/utils/UnitUtils.wurst

```diff
@@ -1,12 +1,14 @@
 package UnitUtils
+import Lodash
 
-public function findNearestUnit(vec2 pos, real range, filterfunc filter) returns unit
```

Comment:
I would keep both versions of this function (let's allow an overload). In fact, to make the interface better, lets add this as well:

```
public function findNearestUnit(vec2 pos, real range, code filter) returns unit
    let filterFunc = Filter(filter)
    GroupEnumUnitsInRange(ENUM_GROUP, pos.x, pos.y, range, filterFunc)
    unit nearest = null
    var bestDist = REAL_MAX
    for u from ENUM_GROUP
        let distSq = pos.distanceToSq(u.getPos())
        if distSq < bestDist and predicate.call(u)
            nearest = u
            bestDist = distSq
    predicate.maybeFree()
    filterFunc.destr()
    return nearest
```

Reply:
What's the difference between a `code` and a `filterfunc`?

Reply:
`code` gets converted to `filterfunc` by passing it to `Filter`

Reply:
I got that from the code. I mean what are the two for? `code` just looks an arbitrary `-> boolean` lambda. Is `Filter` just some arbitrary construct used for `GroupEnumUnitsInRange`?

Reply:
Neither are a lambda, they are both basic types that JASS supports. `code` can return `boolean` or `nothing` where as `Filter` should only take `code` variables which return `boolean`. `Filter` is used in other places as well, but it does seem like a somewhat arbitrary construct. There is more info here: http://jass.sourceforge.net/doc/types.shtml